projects/plugins/jetpack/extensions/plugins/ai-assistant-plugin/components/seo-enhancer/use-seo-requests.ts

```diff
@@ -131,24 +143,45 @@ export const useSeoRequests = (
 		[ request, editPost ]
 	);
 
+	const updateAltText = useCallback(
+		async ( block: Block, useBase64Image: boolean = false ) => {
+			if ( isImageBusy( block.clientId ) ) {
+				debug( 'Already updating alt text, skipping' );
+				return;
+			}
+
+			try {
+				setImageBusy( block.clientId, true );
+				const response = await request( 'images-alt-text', block, useBase64Image );
+				const altText = parseResponse( response ).texts?.[ 0 ];
+				await updateBlockAttributes( block.clientId, { alt: altText } );
+				setImageBusy( block.clientId, false );
+			} catch ( error ) {
+				setImageBusy( block.clientId, false );
+
+				// If the image URL is invalid, try again with a base64 image.
+				if ( error?.message.includes( 'The image URL is invalid' ) && ! useBase64Image ) {
+					debug( 'Retrying with base64 image' );
+					return updateAltText( block, true );
+				}
+
+				debug( 'Error updating alt text', error );
+			}
+		},
+		[ isImageBusy, setImageBusy, request, updateBlockAttributes ]
+	);
+
 	const updateAltTexts = useCallback(
 		async ( force: boolean = false ) => {
 			const imageBlocks = getAllBlocks().filter( block => block.name === 'core/image' );
```

Comment:
This won't get featured images, right? I think it's fine for now, but we'll have to revisit it.
[Edit] The feature image block doesn't have an alt-text input box in the sidebar, so I'm not sure if there's even a way to add the alt text now.

Reply:
Yeah, I think the featured image does not have alt text :thinking: